lib/rubygems/core_ext/kernel_warn.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+if RUBY_VERSION >= "2.5"
+  module Kernel
+    path = "#{__dir__}/"
+    original_warn = instance_method(:warn)
+    Module.new {define_method(:warn, original_warn)}
```

Comment:
This anonymous module is never used, as far as I can tell.

Reply:
To suppress "method redefined" warning when `$VERBOSE`.

Reply:
Is there any reason to prefer this + `define_method` over the typical `alias`-followed-by-`def` approach along the lines of this?:

```ruby
alias original_warn warn

module_function def warn(*messages, uplevel: nil)
  # <...>
end
```

(I ask because that's more in line with how the rest of the RubyGems codebase does it.)

Reply:
@duckinator consider the scope along with module_function.  Or, which method are you aliasing, class or instance?  Did you try your code?

Reply:
Ah, okay, scope explains it &mdash; my version leaks `original_warn`, which is worse inside of `Kernel` than it would be elsewhere. @nobu's approach avoids this problem.

So what @nobu did is definitely the better option here.

@MSP-Greg I did try it, and it functions as intended, but I initially missed the scope-related implications of it being in `Kernel`. Thanks for pointing that out.